### DISCUSSION TITLE
SQL - CREATE VERTEX without IF NOT EXISTS option

### DISCUSSION THREAD

**Discussion by mik1234mc** [ArcadeData/arcadedb#610] [Q&A]
Is there a reason that SQL CREATE VERTEX command does not have IF NOT EXISTS option? The similar commands (CREATE TYPE, CREATE EDGE, CREATE PROPERTY) have it. Is it possible to implemented for CREATE VERTEX as well?

**Comment by lvca** [comment]
Apart from the DDL commands (create type, property, etc) the only exception for `IF NOT EXISTS` is the create edge command. In this particular case, a check between an existent edge from/to is verified beforehand. In the case of create vertex (or document), there is no such concept. The only thing I can think of is checking if some properties are duplicated, but in this case, you can use a unique index on those properties to avoid duplication.
